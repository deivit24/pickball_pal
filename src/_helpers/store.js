import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
