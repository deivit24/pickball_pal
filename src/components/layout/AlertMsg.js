// React
import React from 'react';
// Redux
import { useSelector, useDispatch } from 'react-redux';
/**
 * For displaying that I can yuse a different
 * method to style a component I am going to leave
 * this here
 */
import { makeStyles } from '@material-ui/core/styles';
// Remove alert action
import { removeAlert } from '../../_actions/alert';
// Alert Component
import Alert from '@material-ui/lab/Alert';

const useStyles = makeStyles((theme) => ({
  root: {
    position: 'fixed',
    bottom: '5%',
    left: '5%',
    width: 'auto',
    zIndex: '50',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
  },
}));

const AlertMsg = () => {
  const classes = useStyles();
  const alerts = useSelector((st) => st.alert);
  const dispatch = useDispatch();

  const deleteAlert = (id) => {
    dispatch(removeAlert(id));
  };
  let alertMessage;
  if (alerts !== null && alerts.length > 0) {
    alertMessage = alerts.map((alert) => (
      <div key={alert.id}>
        <Alert
          variant="filled"
          severity={alert.alertType}
          onClose={() => deleteAlert(alert.id)}
        >
          {alert.msg}
        </Alert>
      </div>
    ));
  } else {
    alertMessage = '';
  }

  return <div className={classes.root}>{alertMessage}</div>;
};

export default AlertMsg;
