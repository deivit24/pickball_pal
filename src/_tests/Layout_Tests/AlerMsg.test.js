import React from 'react';
import { render } from '@testing-library/react';
import AlertMsg from '../../components/layout/AlertMsg';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing AlertMsg

describe('Testing the AlertMsg', () => {
  it('it renders AlertMsg without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <AlertMsg />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <AlertMsg />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
