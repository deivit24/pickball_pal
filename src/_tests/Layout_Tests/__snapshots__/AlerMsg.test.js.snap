// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the AlertMsg matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-2"
  />
</DocumentFragment>
`;
